SystemManager/Src/SystemManager.cpp

```diff
@@ -19,25 +19,26 @@ SystemManager::SystemManager():
     pitchMotorChannel_(&htim2, TIM_CHANNEL_4),
     watchdog_(&hiwdg)
 {}
+
 SystemManager::~SystemManager() {}
 
 void SystemManager::flyManually() {
     for(;;){
-            updateRCInputs();
-            executeInputs();
-    }
-}
+        this->rcInputs_ = rcController_.GetRCControl();
+        if (this->rcController_.getIsDataNew()) watchdog_.refreshWatchdog();
 
-void SystemManager::updateRCInputs() {
-    this->rcInputs_.assignValues(rcController_.GetRCControl());
-    watchdog_.refreshWatchdog();
-    return;
-}
-
-void SystemManager::executeInputs() {
-    this->throttleMotorChannel_.set(rcInputs_.throttle);
-    this->yawMotorChannel_.set(rcInputs_.yaw);
-    this->rollMotorChannel_.set(rcInputs_.roll);
-    this->pitchMotorChannel_.set(rcInputs_.pitch);
-    return;
+        if(this->rcInputs_.arm >= (SBUS_MAX - SBUS_THRESHOLD)) {
+            this->throttleMotorChannel_.set(rcInputs_.throttle);
+            this->yawMotorChannel_.set(rcInputs_.yaw);
+            this->rollMotorChannel_.set(rcInputs_.roll);
+            this->pitchMotorChannel_.set(rcInputs_.pitch);
+        }
+        else if (this->rcInputs_.arm <= SBUS_THRESHOLD) {
+            this->throttleMotorChannel_.set(0);
+            this->yawMotorChannel_.set(0);
+            this->rollMotorChannel_.set(0);
+            this->pitchMotorChannel_.set(0);
+        }
```

Comment:
What happens if the arm channel reads between the thresholds? I think this should just be a straight `if {} else {}` scenario

Reply:
this was discussed at the meeting actually, @Chrisc110 mentioned doing it this way in case the arm (or any other binary channel) was some value in between 0 and 100. is there a guaruntee that itll be either one or the other?

Reply:
The values for arm/disarm are binary (either 0 or 100) so we added some threshold in case there is some noise around it. If its in the middle somewhere, we just ignore it I believe

Reply:
My argument is that we should do something in the middle. Right now, the drone will lock up and be both uncontrollable and flying at the exact configuration it was before the flaky arm/disarm signal is received.

Reply:
For context, someone could bind a 3 position switch to the arm/disarm channel (I know Anthony does this, and hopefully he also adjusts the controller settings to only output 0 or 100) so it wouldn't be impossible to accidentally send 50 on the arm/disarm channel.

Reply:
It's setup so that the arm/disarm will only be 0 or 100, but it is possible for a controller to "flake" and read non-0 or 100 values...

This does not stop other people from doing the same thing and forgetting to do the check and sending 0-50-100.

Reply:
I looked at the code a bit more, I think the safest thing you can do is check for arm if ARM > MAX - thresh. Adding a check to disarm doesn't make sense (let's say you lose RC link and the failsafe isn't configured properly, so you output 50% or 70% or whatever the failsafe is set to).